Automate build environment

JochiPochi opened the pull request:
The modifications to the manifest file were inspired by that of the Freescale community BSP.

https://github.com/Freescale/fsl-community-bsp-platform

The script `setup-environment` ensures that both `local.conf` and `bblayers.conf` are set up correctly and it automatically updates to the latest version every time it runs. This ensures that a user can set up a build environment with less steps and with a lot less margin for error.
`setup-environment` lives on meta-intel-aero and it is meant to be used along side the following PR:
https://github.com/intel-aero/meta-intel-aero/pull/10

At the moment, there is an extra commit "Temp commit for testing" that should be removed before merging.
In order to test this PR, simply execute the following commands:
```
mkdir cyphy-aero
cd cyphy-aero
repo init -u https://github.com/cyphyworks/intel-aero-manifest.git
repo sync
source setup-environment integration-test1
bitbake *** anything you want ***
```
Make sure that the environmental variable TEMPLATECONF is not set. as it will cause the configuration to fail.

JochiPochi commented:
Hey @intel-aero, I noticed you guys added a new default.xml not too long ago. Is there anything on this PR that would be useful to you or should I close it?

lucasdemarchi commented:
Thanks, @JochiPochi. I think it's very useful and improves the workflow a lot. Can you rebase this PR?

JochiPochi commented:
Sure thing!

lucasdemarchi commented:
@JochiPochi do you think you can update this soon?   We are going to do a release next week and it would be good to have it in.

JochiPochi commented:
I'm in the middle of testing. I'll be done by the end of the day.

Commit:
Bump defaults.xml to use v01.00.05

JochiPochi commented:
I encountered a minor error while testing, so this is delaying me a little bit, but I'm working on it. I'll keep you posted.

Commit:
Temp commit for testing

Commit:
Add default revision for tag v01.00.04

Commit:
Re-lable projects section to remotes

Commit:
Introduce new build instructions

JochiPochi commented:
@lucasdemarchi I'm testing again now that #10 seems resolved. I'll let you know once this is good for review again.

Commit:
Add "projects" section

The projects that don't have a revision with them will use the default revision
introduced previously:
<default sync-j="4" revision="*******"/>

This simplifies upgrading the manifest file if the projects are tagged properly.

JochiPochi commented:
My last test is still ongoing, I synced, rebased and everything is happy now. 
There is one extra commit "Temp commit for testing" that would like to remove if everything else looks good. You can also give this work a test if you want, I'll be updating the PR description with instructions on how to test in a minute.
